FAQ: should I enable flakes (#547)
NixOS/nix.dev: source/recipes/faq.md

## Nix

### Should I enable flakes?

You have to judge for yourself based on your needs.

Flakes and the `nix` command suite bring multiple improvements that are relevant for both software users and package authors:
- The new command line interface, together with flakes, makes dealing with existing packages significantly more convenient.
- The constraints imposed on flakes strengthen reproducibility by default, and enable various performance improvements when interacting with a large Nix package repository like {term}`Nixpkgs`.
- Flake references allow for easier handling of version upgrades for existing packages or project dependencies.
- The flake schema helps with composing Nix projects from multiple sources in an ordered fashion.

Other than that, and below the surface of the flake schema, Nix and the Nix language work the exactly same in both cases.

> The same level of reproducibility can be achieved  with or without flakes.

Author: ```suggestion
In principle, same level of reproducibility can be achieved with or without flakes.
```

@zmitchell given the significant increase in convenience is stated at the beginning, this should be acceptable. What do you think?

Reviewer: ```suggestion
In principle, the same level of reproducibility can be achieved  with or without flakes.
```

That sounds fine, you're just missing a word here.

> Nix offers the same reproducibility guarantees with or without flakes, when used correctly.

Reviewer: Suggestion: "correctly" is a matter of opinion, no? A better phrasing may be "The same level of reproducibility is possible with or without flakes."

Author: I wanted to somehow express the caveat that you have to make certain precautions without flakes, such as avoiding search paths - essentially what have been called anti patterns so far. It would be unfair to say that classic Nix is just as reproducible as flakes by default: you have to do something consciously to get there.

Author: ```suggestion
The same level of reproducibility can be achieved  with or without flakes.
```
